tools/nnyeah/nnyeah/ModuleExtensions.cs

```diff
@@ -4,23 +4,46 @@
 
 namespace Microsoft.MaciOS.Nnyeah {
 	public static class ModuleExtensions {
-		static Dictionary<string, PlatformName> moduleToPlatform = new Dictionary<string, PlatformName> {
+		static Dictionary<string, PlatformName> xamarinModuleToPlatform = new Dictionary<string, PlatformName> {
 			{ "Xamarin.iOS", PlatformName.iOS },
 			{ "Xamarin.Mac", PlatformName.macOS },
 			{ "Xamarin.TVOS", PlatformName.tvOS },
 			{ "Xamarin.WatchOS", PlatformName.watchOS }
 		};
 
+		static Dictionary<string, PlatformName> microsoftModuleToPlatform = new Dictionary<string, PlatformName> {
+			{ "Microsoft.iOS", PlatformName.iOS },
+			{ "Microsoft.macOS", PlatformName.macOS },
+			{ "Microsoft.tvOS", PlatformName.tvOS },
+			{ "Microsoft.watchOS", PlatformName.watchOS }
+		};
+
 		public static bool DependsOnXamarin (this ModuleDefinition module)
 		{
 			return module.XamarinPlatformName () != PlatformName.None;
 		}
 
 		public static PlatformName XamarinPlatformName (this ModuleDefinition module)
+		{
+			return FindPlatformName (module, xamarinModuleToPlatform);
+		}
+
+		public static bool DependsOnMicrosoft (this ModuleDefinition module)
+		{
+			return module.MicrosoftPlatformName () != PlatformName.None;
+		}
+
+		public static PlatformName MicrosoftPlatformName (this ModuleDefinition module)
+		{
+			return FindPlatformName (module, microsoftModuleToPlatform);
+		}
+
+		static PlatformName FindPlatformName (ModuleDefinition module, Dictionary<string, PlatformName> knownModules)
```

Comment:
why do you need to dictionaries? The keys are different, so you just simply add all the keys and some eyes return the same value, no need for two collection, no need for the wrapping methods, it will work the same with less indirection and less code.

@chamons  though?

Reply:
Yes, there's a very good reason. If I ask what the _Xamarin_ platform is, if I get `Platform.None` it means there is no Xamarin platform. If I make the change and I ask what the Xamarin platform is, it _may_ give me the Microsoft platform, which is decidedly not what was asked.